File: client/js/npcs.js

define(['npc'], function(Npc) {

    var NPCs = {

        Guard: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.GUARD, 1);
                this._name = "<Guard />";
            }
        }),

        King: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.KING, 1);
                this._name = "King";
            }
        }),

        Agent: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.AGENT, 1);
                this._name = "Agent Smith";
            }
        }),

        Rick: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.RICK, 1);
                this._name = "Rick";
            }
        }),

        VillageGirl: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.VILLAGEGIRL, 1);
                this._name = "Village Girl";
            }
        }),

        Villager: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.VILLAGER, 1);
                this._name = "Villager";
            }
        }),
        
        Coder: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.CODER, 1);
                this._name = "Coder";
            }
        }),

        Scientist: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.SCIENTIST, 1);
                this._name = "Scientist";
            }
        }),

        Nyan: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.NYAN, 1);
                this.idleSpeed = 50;
                this._name = "Nyan Cat";
            }
        }),
        
        Sorcerer: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.SORCERER, 1);
                this.idleSpeed = 150;
                this._name = "Sorcerer";
            }
        }),

        Priest: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.PRIEST, 1);
                this._name = "Priest";
            }
        }),
        
        BeachNpc: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.BEACHNPC, 1);
                this._name = "Surfer";
            }
        }),
        
        ForestNpc: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.FORESTNPC, 1);
                this._name = "Forest Keeper";
            }
        }),

        DesertNpc: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.DESERTNPC, 1);
                this._name = "Traveler";
            }
        }),

        LavaNpc: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.LAVANPC, 1);
                this._name = "Geologist";
            }
        }),

        Octocat: Npc.extend({
            init: function(id) {
                this._super(id, Types.Entities.OCTOCAT, 1);
                this._name = "Octocat";
            }
        })
    };
    
    return NPCs;
});

File: client/dart/npcs.dart
library npcs;

import "npc.dart";
import "../shared/dart/gametypes.dart";

class Guard extends Npc {

  Guard(int id): super(id, Entities.GUARD, "<Guard />");
}

class King extends Npc {

  King(int id): super(id, Entities.KING, "King");
}

class Agent extends Npc {

  Agent(int id): super(id, Entities.AGENT, "Agent Smith");
}

class Rick extends Npc {

  Rick(int id): super(id, Entities.RICK, "Rick");
}

class VillageGirl extends Npc {

  VillageGirl(int id): super(id, Entities.VILLAGEGIRL, "Village Girl");
}

class Villager extends Npc {

  Villager(int id): super(id, Entities.VILLAGER, "Villager");
}

class Coder extends Npc {

  Coder(int id): super(id, Entities.CODER, "Coder");
}

class Scientist extends Npc {

  Scientist(int id): super(id, Entities.SCIENTIST, "Scientist");
}

class Nyan extends Npc {

  Nyan(int id): super(id, Entities.NYAN, "Nyan Cat");
}

class Sorcerer extends Npc {

  Sorcerer(int id): super(id, Entities.SORCERER, "Sorcerer");
}

class Priest extends Npc {

  Priest(int id): super(id, Entities.PRIEST, "Priest");
}

class Surfer extends Npc {

  Surfer(int id): super(id, Entities.BEACHNPC, "Surfer");
}

class ForestKeeper extends Npc {

  ForestKeeper(int id): super(id, Entities.FORESTNPC, "Forest Keeper");
}

class Traveler extends Npc {

  Traveler(int id): super(id, Entities.DESERTNPC, "Traveler");
}

class Geologist extends Npc {

  Geologist(int id): super(id, Entities.LAVANPC, "Geologist");
}

class Octocat extends Npc {

  Octocat(int id): super(id, Entities.OCTOCAT, "Octocat");
}
